packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
 const policyUtils = require('strapi-utils').policy;
 
 module.exports = {
+  /**
+   * Returns all fields of type primitive
+   * 
+   * @returns {Boolean}
+   */
+  isPrimitiveType: (_type) => {
+    const type = _type.replace('!', '');
+    return (
+      type === 'Int' ||
+      type === 'Float' ||
+      type === 'String' ||
+      type === 'Boolean'
+    );
+  },
+
+  /**
+   * Returns all fields that are not of type array
+   * 
+   * @returns {Boolean}
+   * 
+   * @example
+   * 
+   * isNotOfTypeArray([String])
+   * // => false
+   * isNotOfTypeArray(String!)
+   * // => true
+   */
+  isNotOfTypeArray: (_type) => {
+    const arrayRegexp = /(\[\w+\])/;
+    const type = _type.replace('!', '');
```

Comment:
Update the regex with `(\[\w+!?\])` to remove the potential useless `.replace()`